### PR TITLE
Get rid of unique suit storage whitelists

### PR DESCRIPTION
**Pull request by sippykot**
## About The Pull Request
Removes unique suit storage whitelists from (almost) every suit and puts them into one big whitelist that combines them.
Modifies files pertaining to clothing using the suit slot, and MODs.
Doesn't add to the total combined list since it's out of scope of this PR; if this is accepted then it's time to think about what to add/remove from the list.
Could be changed to have some more restrictions/whitelist types.
## Why It's Good For The Game
Current system of suit storage whitelists is arbitrary and restricting. - some suits can carry some items, some don't some only certain types, such as only one kind of bag but not a different kind of bag.  It doesn't make much sense either from gameplay or flavour side of things. Changing it to one/few lists makes it less confusing, less restricting in what you can wear and easier to maintain/rebalance.
## Changelog
Removes unique suit storage whitelists from (almost) every suit and puts them into one big whitelist that combines them.
:cl:
/:cl:
(haven't added tags since im not sure what to add, i tested it and it does compile and work so ill put it as a normal PR, hopefully i haven't missed anything)

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
Ah yes, the fabled instant merge conflict.

**Comment by ZephyrTFA** [comment]
Also I don't agree with this.
It makes thematic and gameplay sense that certain outerwear can only hold similarly themed items.

**Comment by sippykot** [comment]
> Also I don't agree with this. It makes thematic and gameplay sense that certain outerwear can only hold similarly themed items.

I don't feel like it's very thematic that you for example can't put on a bag because it's the wrong colour...

**Comment by ZephyrTFA** [comment]
Thats a trick statement. There is no restrictions on the back slot when considering outerwear (except the modsuit, we dont talk about the modsuit)
🥹

**Comment by sippykot** [comment]
> Thats a trick statement. There is no restrictions on the back slot when considering outerwear (except the modsuit, we dont talk about the modsuit) 🥹

i was talking about the suit storage slot, obviously :/
but yeah i don't get why it's "thematic", just seems more arbitrary and annoying to me

**Comment by ZephyrTFA** [comment]
I think its cool that you can sometimes only use matching attire. Not everything needs to be unfun 😢

**Comment by MrMelbert** [comment]
There's certainly some instances I disagree with this.
- Guns/holsters
- Mining gear
- Swords
- Spears(!!!)

The allowed list is just one way we encourage people to wear clothing befitting their job, and while imperfect in some instances (the aforementioned bag on bag thing) it does do the job

**Comment by Blacklist897** [comment]
the magharness allows mods that cant normaly have guns to carry guns

File: code/modules/clothing/suits/_suits.dm [hunk 1]
```diff
@@ -4,18 +4,172 @@
...
-		/obj/item/tank/jetpack/oxygen/captain,
-		/obj/item/storage/belt/holster,
-		)
 	armor_type = /datum/armor/none
 	drop_sound = 'sound/items/handling/cloth_drop.ogg'
 	pickup_sound = 'sound/items/handling/cloth_pickup.ogg'
 	slot_flags = ITEM_SLOT_OCLOTHING
 	var/blood_overlay_type = "suit"
 	limb_integrity = 0 // disabled for most exo-suits
+	allowed = list(
```
Comment (MrMelbert) [inline code/modules/clothing/suits/_suits.dm:13]: Also - no matter what happens, you should just outright delete the `allowed` var and have a global typecache. There's no need to have this massive list on every single suit item

Comment (MrMelbert) [reply code/modules/clothing/suits/_suits.dm:13]: Or I guess if it's still used in some places, then put this mega-list in a global typecache and check `allowed || GLOB.suit_allowables` wherever relevant

**Comment by sippykot** [comment]
> There's certainly some instances I disagree with this.
> 
> * Guns/holsters
> * Mining gear
> * Swords
> * Spears(!!!)
> 
> The allowed list is just one way we encourage people to wear clothing befitting their job, and while imperfect in some instances (the aforementioned bag on bag thing) it does do the job

Honestly i agree that there could be some restrictions on the guns/holsters. As i described this is taking everything from the lists and merging them together into one. Could make a more restricted version and have an extra list that adds these to some suits... But i don't really see it as a big issue to be honest. And when it comes to job clothing it's usually either that or straight up armor as the "best" option stats wise either way.

File: code/modules/clothing/suits/_suits.dm (hunk 1 above)
Comment (sippykot) [reply code/modules/clothing/suits/_suits.dm:13]: maybe... im pretty new to byond stuff but i could look into it

Comment (MrMelbert) [reply code/modules/clothing/suits/_suits.dm:13]: Ask if you need help but as it stands putting a 150 length element list on every suit item is a bit of a waste of memory

Comment (sippykot) [reply code/modules/clothing/suits/_suits.dm:13]: yeah, id appreciate help on that x3
and im open to nerfing the gun stuff, it's not really the main point of the PR

**Comment by MrMelbert** [comment]
> But i don't really see it as a big issue to be honest.

Spears alone are a huge deal breaker to me - their made trade off is they are bulky and can't be stored easily (normally requiring a back slot), slapping it on any suit storage is scary

> And when it comes to job clothing it's usually either that or straight up armor as the "best" option stats wise either way.

The thing is it's accomplishing exactly what we want, IE encouraging people wear their departmental clothing.

If you want to store an extra 50 plants on your person you wear a botanist smock, if you want to store more chems you wear a chemist coat.

With this PR, there's 0 incentive to not wear the strongest armor you can get your hands on.

**Comment by sippykot** [comment]
> > But i don't really see it as a big issue to be honest.
> 
> Spears alone are a huge deal breaker to me - their main trade off is "they are bulky and can't be stored easily" (normally requiring a back slot). Slapping it on just any suit storage is scary
> 
> > And when it comes to job clothing it's usually either that or straight up armor as the "best" option stats wise either way.
> 
> The thing is it's accomplishing exactly what we want, IE encouraging people wear their departmental clothing.
> 
> If you want to store an extra 50 plants on your person you wear a botanist smock, if you want to store more chems you wear a chemist coat.
> 
> With this PR, there's 0 incentive to not wear the strongest armor you can get your hands on.

there's 0 incentive to not wear the strongest armor if you want to fight either - i agree that the spears/guns might be too strong and im open to changing it

right now you have to dress up as the wrong department/job if you want extra bit of storage on you, like if a chef wants to carry a plant bag or chem bag with beakers in the suit slot, they have to dress up as chemist or botanist, etc

**Comment by Iamgoofball** [comment]
At this point, delete suit storage and just add storage components to the relevant items

**Comment by ZephyrTFA** [comment]
Goof, stop cooking on a thursday.
You know we only have it on tuesdays.

**Comment by optimumtact** [comment]
Your arguments for why this should be approved are not particularly compelling.
 
 These restrictions are not arbitrary they're on purpose.

**Comment by sippykot** [comment]
> Your arguments for why this should be approved are not particularly compelling.
> 
> These restrictions are not arbitrary they're on purpose.

What is the purpose of restricting what you can carry in that one specific slot? It does just incentivize not dressing as your job because you want to put something specific in there. Or things like command not wearing winter jackets because you can't put an e-gun in them.

**Comment by Fikou** [comment]
> What is the purpose of restricting so heavily what you can carry in that one specific slot? 

it allows a balancing lever of control over what you hold to be appropriate for the suit, instead of having any suit be a super convenient gun axe spear jetpack holder. 
> It does just incentivize not dressing as your job because you want to put something specific in there.

i dont find this accurate at all, way more i have found that i am incentivized to dress as my job because i have a job related item that i want to hold conveniently in the suit storage. if i could put it anywhere i would have less reason to do this
> Or things like command not wearing winter jackets because you can't put an e-gun in them.

then open a pr to make the suits that should hold something but dont hold them

**Pull request closed without merging**